tasks/apt.yml

```diff
@@ -37,10 +37,7 @@
 - name: Set openwisp2_spatialite_path
   set_fact:
     openwisp2_spatialite_path: false
-  when: >
-    openwisp2_database.engine != "django.contrib.gis.db.backends.spatialite"
-    or (openwisp2_spatialite_path == 'mod_spatialite'
-        and mod_spatialite_result is failed)
+  when: mod_spatialite_result is failed
```

Comment:
Made all the changes except one:
I have not removed the following code yet:
```
  register: mod_spatialite_result

- name: Set openwisp2_spatialite_path
  set_fact:
    openwisp2_spatialite_path: false
  when: >
    openwisp2_database.engine != "django.contrib.gis.db.backends.spatialite"
    or (openwisp2_spatialite_path == 'mod_spatialite'
        and mod_spatialite_result is failed)
```
I am thinking since the code is still relevant in case user tries to install in older ubuntu / debian versions.
Example: Ubuntu
since `openwisp2_spatialite_path: "mod_spatialite"` is set for version less than 18.04, 
older versions like 12.04 will still fail and would require `openwisp2_spatialite_path: false` to be set to false.

However, i have modified the condition, since `openwisp2_database.engine != "django.contrib.gis.db.backends.spatialite"` doesn't need to changed again (as it was checked in the task above) and `openwisp2_spatialite_path` could have different values.